Assets/Scripts/VR/VRInteraction.cs

```diff
@@ -65,38 +80,52 @@ void showIndicator(GameObject controllerObject, bool showIndicator)
             if (m_InputModule != null && m_InputModule.rayTransform != null) rayOrigin = m_InputModule.rayTransform;
             laserStartPos = rayOrigin.position;
             forwardDirection = rayOrigin.forward;
-            laserEndPos = laserStartPos + (laserMaxLength * forwardDirection);
+            laserEndPos = laserStartPos + (laserLongDistance * forwardDirection);
             updateLaser(false);
-            // now detect if the user is holding down the Interact button also
+            // Next, detect if the user is holding down the Interact button also
 
-            // raycast, then if we hit the Earth we can show the interaction
-            ray = new Ray(laserStartPos, forwardDirection);
-            if (Physics.Raycast(ray, out hit, laserMaxLength, layerMask))
+            if (mainUIController == null) mainUIController = FindObjectOfType<MainUIController>();
+            // if we're drawing, no need to raycast
+            if (mainUIController.IsDrawing && annotationTool != null)
+            {
+                if (interactionTrigger())
+                {
+                    annotationTool.Annotate(laserEndPos);
+                }
+
+            }
+            else
             {
-                laserEndPos = hit.point;
-                updateLaser(true);
+                // Raycast, then if we hit the Earth or a star we can show the interaction
+                ray = new Ray(laserStartPos, forwardDirection);
 
-                if (hit.transform.name == "Earth")
+                if (Physics.RaycastNonAlloc(ray, hits, laserShortDistance, layerMaskEarth) > 0)
                 {
-                    if (interactionTrigger())
+                    for (int i = 0; i < hits.Length; i++)
                     {
-                        if (canInteract)
-                        {
-                            // filter by what we hit
+                        hit = hits[i];
+                        laserEndPos = hit.point;
+                        updateLaser(true);
 
-                            manager = SimulationManager.GetInstance();
-                            interactionController = FindObjectOfType<InteractionController>();
-                            interactionController.ShowEarthMarkerInteraction(hit.point, Quaternion.FromToRotation(Vector3.forward, hit.normal), manager.LocalPlayerColor, true);
-                            canInteract = false;
+                        if (interactionTrigger())
+                        {
+                            Collider c = hit.collider;
+                            if (c is SphereCollider)
```

Comment:
If the user 'clicks' on the image of earth in VR, Is it true that at least two hits will come back. One for the Sphere, and one for the mesh?

Reply:
Correct! the Earth now has two colliders, one that is a perfect sphere, the other conforms to the mesh. The Mesh collider has knowledge of the underlying texture used for the planet. When we RaycaseNonalloc it's doing a multiple raycast and we gain understanding of all the things that the rays hit.